lib/temporal/testing/scheduled_workflows.rb

```diff
@@ -0,0 +1,55 @@
+module Temporal
+  module Testing
+    class WorkflowIDNotScheduled < ClientError; end
+
+    # Implementation for Temporal::Testing::ScheduledWorkflwows
+    module ScheduledWorkflowsImpl
+      class << self
+
+        # For someone who wants to assert that the schedule is what they expect.
+        # Populated by Temporal.schedule_workflow
+        # format: { <workflow_id>: <cron schedule string>, ... }
+        def schedules
+          _schedules.dup.freeze
```

Comment:
This will create a new object every time it is accessed, which seems wasteful. Why not do this in a functional way instead — create a new frozen hash every time you add a value?